cupyx/scipy/ndimage/filters.py

```diff
@@ -969,381 +993,147 @@ def _get_rank_kernel(filter_size, rank, mode, wshape, origins, cval, int_type):
-        _check_origin(origin, width)
-    return tuple(origins), int_type
-
-
-def _run_1d_filters(filters, input, args, output, mode, cval, origin=0):
-    """
-    Runs a series of 1D filters forming an nd filter. The filters must be a
-    list of callables that take input, arg, axis, output, mode, cval, origin.
-    The args is a list of values that are passed for the arg value to the
-    filter. Individual filters can be None causing that axis to be skipped.
+    Args:
+        input (cupy.ndarray): The input array.
+        function (cupy.ReductionKernel, cupy.RawKernel,
+            cupy.core.fusion.Fusion, or callable):
+            The kernel or function to apply to each region.
+        size (int or sequence of int): One of ``size`` or ``footprint`` must be
+            provided. If ``footprint`` is given, ``size`` is ignored. Otherwise
+            ``footprint = cupy.ones(size)`` with ``size`` automatically made to
+            match the number of dimensions in ``input``.
+        footprint (cupy.ndarray): a boolean array which specifies which of the
+            elements within this shape will get passed to the filter function.
+        output (cupy.ndarray, dtype or None): The array in which to place the
+            output. Default is is same dtype as the input.
+        mode (str): The array borders are handled according to the given mode
+            (``'reflect'``, ``'constant'``, ``'nearest'``, ``'mirror'``,
+            ``'wrap'``). Default is ``'reflect'``.
+        cval (scalar): Value to fill past edges of input if mode is
+            ``'constant'``. Default is ``0.0``.
+        origin (scalar or tuple of scalar): The origin parameter controls the
+            placement of the filter, relative to the center of the current
+            element of the input. Default of 0 is equivalent to
+            ``(0,)*input.ndim``.
+    Returns:
+        cupy.ndarray: The result of the filtering.
+
+    .. notes::
+        If the `function` is a :class:`cupy.RawKernel` then it must be for a
+        function that has the following signature. Unlike most functions, this
+        should not utilize `blockDim`/`blockIdx`/`threadIdx`.
+
+            __global__ void func(double *buffer, int filter_size,
+                                 double *return_value)
+
+        If the `function` is a :class:`cupy.ReductionKernel` then it must be
+        for a kernel that takes 1 array input and produces 1 'scalar' output.
+
+        If the `function` is a callable, it must be fuseable. If it is a fused
+        function it must either:
+
+          * resolve to a :class:`cupy.ReductionKernel` that meets the
+            requirements above
+          * resolve to a `FusedKernel` that takes 1 array input argument and
+            returns 1 'scalar' output
+
+    .. seealso:: :func:`scipy.ndimage.generic_filter`
     """
-    output_orig = output
+    from cupyx.scipy.ndimage._filters_generic import (
```

Comment:
I think the CuPy style may require moving this to the top of the file and using

`from cupyx.scipy.ndimage import _filters_generic`
and then calling via `_filters_generic._get_generic_filter_raw`, etc.